src/types/routeMethods.ts

```diff
@@ -71,12 +71,12 @@ type UnifyParamEnds<
 
 export type ExtractParamsFromPathString<
   TPath extends string,
-  TParams extends Record<string, Param> = Record<never, never>
-> = TPath extends `${infer Path}${ParamEnd}`
+  TParams extends Record<string, Param | undefined> = Record<never, never>
+> = UnifyParamEnds<TPath> extends `${infer Path}${ParamEnd}`
   ? ExtractParamsFromPathString<Path, TParams>
-  : TPath extends `${string}:${infer Param}${ParamEnd}${infer Rest}`
+  : UnifyParamEnds<TPath> extends `${string}:${infer Param}${ParamEnd}${infer Rest}`
     ? MergeParams<{ [P in ExtractParamName<Param>]: ExtractPathParamType<Param, TParams> }, ExtractParamsFromPathString<Rest, TParams>>
-    : TPath extends `${string}:${infer Param}`
+    : UnifyParamEnds<TPath> extends `${string}:${infer Param}`
```

Comment:
I moved wrapping the path in `UnifyParamEnds` into this type so that we don't have to remember wrap any path string we pas into this type in it.